src/transformers/models/llava/processing_llava.py

```diff
@@ -51,6 +51,9 @@ class LlavaProcessor(ProcessorMixin):
             The tokenizer is a required input.
         patch_size (`int`, *optional*):
             Patch size from the vision tower.
+        num_additional_image_tokens (`int`, *optional*, defaults to 0):
+            Number of additional tokens added to the image embeddings, such as CLS (+1). If the backbone has no CLS or other
```

Comment:
they are added where (to which image embedding and why) also I am not even sure I understand the usage of this myself. As it will be in all processors, can you explain a bit more why someone should set this?

Reply:
Yes, the very first reason for this PR is that some ViT backbones can add a CLS token to image patches, while other do not. Therefore our current processors can't work with SigLIP because we hardcoded the CLS token addition in code.

We had two options to fix it:
- what I proposed initially was to simply use `num_image_tokens` arg and let users specify any amount so we don't have to infer how many patches there will be from `patch_size` and `image_size`. Some processor already do that like Paligemma
- it had drawbacks though that we cannot work with non-square images in that case, because in case of llava-next the padding/unpadding is needed. That depends on `patch-size` and `image-size`. I realize we have no models that accept non-square images, but that was the strongest objection from core maintainer against the `num-image-tokens`

So after discussions with Pavel, we decided to make as less changes as possible and still support SigLIP by adding `num_additional_tokens`. It will be 1 if the vision tower adds CLS, and otherwise 0. It can be more than 1 in case there is any vision tower with extra tokens added on top of image patches

Reply:
ok thanks for explaning!